### PR TITLE
vm tests: update and deprcated OS's tested

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/38009

This changes the set of OS's we test to improve coverage while decreasing test count.

Before merging, I will need to build the new bases; I put a hold to do that. Will do so after approval.

There were a number of related fixes to make this (easily) work; comments inline describing why.

Testing: we only test all of these in postsubmit; I manually tested.

### CODE REVIEW

File: pkg/test/framework/components/echo/kube/deployment.go [hunk 1]
```diff
@@ -613,14 +613,20 @@ func GenerateService(cfg echo.Config) (string, error) {
...
-	echo.Debian9:      "app_sidecar_debian_9",
-	echo.Debian10:     "app_sidecar_debian_10",
+	echo.UbuntuJammy:  "app_sidecar_ubuntu_jammy",
+	echo.Debian11:     "app_sidecar_debian_11",
 	echo.Centos7:      "app_sidecar_centos_7",
-	echo.Centos8:      "app_sidecar_centos_8",
+	echo.Rockylinux8:  "app_sidecar_rockylinux_8",
 }
 
+var RevVMImages = func() map[string]echo.VMDistro {
```
Comment (howardjohn) [inline pkg/test/framework/components/echo/kube/deployment.go:622]: I think previously we just always tested DefaultVMDistro repeatedly... fixing this

File: tools/build-base-images.sh [hunk 2]
```diff
@@ -23,13 +23,23 @@ ROOT=$(dirname "$WD")
...
+}
+
 # shellcheck source=prow/lib.sh
 source "${ROOT}/prow/lib.sh"
 buildx-create
 
 HUBS="${HUBS:?specify a space seperated list of hubs}"
 TAG="${TAG:?specify a tag}"
-DOCKER_TARGETS="${DOCKER_TARGETS:-base,distroless,app_sidecar_base_debian_9,app_sidecar_base_debian_10,app_sidecar_base_ubuntu_xenial,app_sidecar_base_ubuntu_bionic,app_sidecar_base_ubuntu_focal,app_sidecar_base_centos_7,app_sidecar_base_centos_8}"
+defaultTargets="$(cat "${ROOT}/tools/docker.yaml" | toJson | toJson | jq '[.images[] | select(.base) | .name] | join(",")' -r)"
```
Comment (howardjohn) [inline tools/build-base-images.sh:41]: auto-derive

File: tools/docker-builder/main.go [hunk 3]
```diff
@@ -126,6 +126,27 @@ func CopyInputs(a Args) error {
 	return nil
 }
 
+func ReadPlanTargets() ([]string, []string, error) {
```
Comment (howardjohn) [inline tools/docker-builder/main.go:129]: So we can automatically derive default targets

**Review by ericvn — APPROVED** [review]

**Pull request merged**